content/docs/interface/example1/_index.md

```diff
@@ -196,55 +215,64 @@ finalise(handle)
 
 ### `link_read()`
 
-- responsible for returning the path of an external object in the local data store
+*responsible for returning the path of an external object in the local data store*
+
 - updates `handle` with file path (if not already present) and useful metadata
 - returns file path
 
 ### `read_array()`
 
-- responsible for reading the correct data product, which at this point has been downloaded from the remote data store by `fair pull`
+*responsible for reading the correct data product, which at this point has been downloaded from the remote data store by `fair pull`*
+
 - reads a specified version of the data
 - updates `handle`
 
 ### `link_write()`
 
-- returns a path to write external objects to
-- updates `handle`
+- updates `handle` with useful metadata
+- returns a path to write to
 
 ### `write_array()`
 
-- responsible for writing an array as a component to an hdf5 file
+*responsible for writing an array as a component to an hdf5 file*
+
+- writes component to the hdf5 file
 - updates `handle`
   - if this is the first component to be written, update `handle` with storage location
   - if this is not the first component to be written, reference the storage location from the `handle`
   - if the component is already recorded in the handle, return the index of this handle reference invisibly
-- writes component to the hdf5 file
 
-### `issue_with_component()`
+### `raise_issue()`
+
+*responsible for writing issue related metadata to the handle*
 
-- **this is very much a work in progress**
-- find the input or output reference (in the handle) that the issue is associated with
-- note that issues can also be associated with scripts, etc. (I've not gone near this yet)
-- record issue metadata in the handle
-- NOTE that in the above example, `issue_with_component()` takes an `index` that references an object recorded in the handle, alternatively it may take a `dataproduct`, `component`, and `version` as identifiers
-- NOTE that `issue_with_component()` is but one of a series of functions including `inssue_with_dataproduct()`, `issue_with_externalobject()`, and `issue_with_script()`; it might make more sense for you to write a generic `raise_issue()` function depending on language constraints
+- records issue and data product / component related metadata in the handle
+- note that where an issue is associated with an entire object, the `whole_object` component is referenced; the `whole_object` component is generated automatically whenever a new Object is created
+- data products or components may be referenced explicitly or via an index (invisibly output from `write_array()` for example)
+- multiple components or data products may be linked to a single issue
```

Comment:
I'm not sure how you do this, but it's probably not the only possible way:
```suggestion
- components or whole data products may be referenced by name or via a reference (for instance returned by `write_array()`)
- multiple components or data products may be linked to a single issue, either via two functions - one to raise issues and one to attach them - or via arguments to `raise_issue()` that allow multiple components to be attached
```

Reply:
It's not the only possible way, but it's an _example_. I did it this way because you asked me to.

Reply:
I don't mind how people do it - I wasn't meaning to force you to do it one particular way, I was just suggesting one possible solution. Anyway, hopefully this edit makes it clear that the other developers have options?

Reply:
Or have you merged this separately without accepting this suggestion?

Reply:
I clicked on the commit suggestion button from your previous edit.

Reply:
Also, I don't know what you mean by:

> via two functions - one to raise issues and one to attach them

since I'm currently raising issues via arguments to `raise_issue()` that allow multiple components to be attached, which are then attached in `finalise()`

Reply:
Hmm, weird. I thought it showed up as accepted when you did that.